crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -381,6 +386,64 @@ where
             )
             .into());
         }
+
+        for receipt in result.receipts() {
+            match receipt {
+                Receipt::Mint {
+                    sub_id,
+                    contract_id,
+                    val,
+                    ..
+                } => {
+                    let asset_id = contract_id.asset_id(sub_id);
+                    let current_count = match db.storage::<AssetsInfo>().get(&asset_id) {
+                        Ok(count) => count,
+                        Err(_) => {
+                            // If asset doesn't exist yet, create it with 0 count
+                            db.storage::<AssetsInfo>()
+                                .insert(&asset_id, &(*contract_id, **sub_id, 0))?;
+                            Some(Cow::Owned((*contract_id, **sub_id, 0)))
+                        }
+                    }
+                    .map(|info| {
+                        info.2
+                            .checked_add(*val)
+                            .ok_or(anyhow::anyhow!("Asset count overflow"))
```

Comment:
You may consider integrating this with the `IndexationError` available here: https://github.com/FuelLabs/fuel-core/blob/master/crates/fuel-core/src/graphql_api/indexation.rs#L28

It's a common type for all errors related to integration and it'll grow even more variants after the "coins to spend" indexation is merged. See [here](https://github.com/FuelLabs/fuel-core/pull/2463/files#diff-b6049ace8304a0ac864e77b0cc1d40c8501613ff153eb7ac0a66a0bc191134de).